src/vs/code/electron-sandbox/issue/issueReporterServiceNew.ts

```diff
@@ -0,0 +1,508 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { $, reset, windowOpenNoOpener } from 'vs/base/browser/dom';
+import { mainWindow } from 'vs/base/browser/window';
+import { Codicon } from 'vs/base/common/codicons';
+import { groupBy } from 'vs/base/common/collections';
+import { CancellationError } from 'vs/base/common/errors';
+import { isMacintosh } from 'vs/base/common/platform';
+import { ThemeIcon } from 'vs/base/common/themables';
+import { URI } from 'vs/base/common/uri';
+import { IssueReporterData as IssueReporterModelData } from 'vs/code/browser/issue/issueReporterModel';
+import { BaseIssueReporterService, hide, show } from 'vs/code/browser/issue/issue';
+import { localize } from 'vs/nls';
+import { isRemoteDiagnosticError } from 'vs/platform/diagnostics/common/diagnostics';
+import { IIssueMainService, IssueReporterData, IssueReporterExtensionData, IssueReporterWindowConfiguration, IssueType } from 'vs/platform/issue/common/issue';
+import { INativeHostService } from 'vs/platform/native/common/native';
+import { applyZoom, zoomIn, zoomOut } from 'vs/platform/window/electron-sandbox/window';
+
+// GitHub has let us know that we could up our limit here to 8k. We chose 7500 to play it safe.
+// ref https://github.com/microsoft/vscode/issues/159191
+const MAX_URL_LENGTH = 7500;
+
+
+export class IssueReporterNew extends BaseIssueReporterService {
+	constructor(
+		private readonly configuration: IssueReporterWindowConfiguration,
+		@INativeHostService private readonly nativeHostService: INativeHostService,
+		@IIssueMainService issueMainService: IIssueMainService
+	) {
+		super(configuration.disableExtensions, configuration.data, configuration.os, configuration.product, mainWindow, false, issueMainService);
+
+		this.issueMainService.$getSystemInfo().then(info => {
```

Comment:
is there anything that waits on this promise to finish in its initialization steps? I see that nothing is waiting on it, but it seems pretty important?

Reply:
I could throw it in a `try...catch` but i think it does the job since `then()` handles the resolved value? `this.updateSystemInfo()` and `this.updatePreviewButtonState()` don't depend on each other or on completion of this promise.